packages/nvidia_nat_core/src/nat/runtime/session.py

```
@@ -474,74 +464,52 @@ async def session(self,
         if isinstance(http_connection, Request):
+            if user_id is None:
+                user_info = UserManager.extract_user_from_connection(http_connection)
+                if user_info is not None:
+                    user_id = user_info.get_user_id()
             token_workflow_parent_id, token_workflow_parent_name = \
                 await self.set_metadata_from_http_request(http_connection)
 
+        if user_id is None and self._is_workflow_per_user:
+            raise ValueError("user_id is required for per-user workflow but could not be determined. "
+                             "Provide a valid JWT in the Authorization header, "
+                             f"set a '{SESSION_COOKIE_NAME}' cookie, or pass user_id explicitly.")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)